app/src/main/java/com/sdpteam/connectout/event/nearbyEvents/list/EventsAdapter.java

```diff
@@ -58,12 +65,23 @@ public View getView(int position, @Nullable View convertView, @NonNull ViewGroup
         eventDescription.setText(event.getDescription());
 
         ImageButton profileImageButton = view.findViewById(R.id.event_list_profile_button);
-        profileImageButton.setOnClickListener(v -> ProfileActivity.openProfile(getContext(), event.getOrganizer()));
+        loadOrganizerProfileImage(orgProfileId, profileImageButton);
+        profileImageButton.setOnClickListener(v -> ProfileActivity.openProfile(getContext(), orgProfileId));
 
-        ImageView eventImage = view.findViewById(R.id.event_list_event_image);
-        //TODO set images
+        ImageView eventImage = view.findViewById(R.id.event_list_event_image);//TODO set event image
         view.setOnClickListener(v -> EventActivity.openEvent(getContext(), event.getId()));
         return view;
     }
+
+    private static void loadOrganizerProfileImage(String organizerProfileId, ImageButton profileImageButton) {
+        // start an async job to load and display the profile image
+        new ProfileFirebaseDataSource().fetchProfile(organizerProfileId).thenAccept(profile -> {
```

Comment:
I feel like it would be better to have a single instance of ProfileFirebaseDataSource instead of creating a new one every time the function is called. Is there a reason that you implemented it this way?

Reply:
There is no need to have one shared instance. We can create as many instances we want because the class is immutable so this is not an issue at all.

Reply:
ok thats great